src/cache/cache_las.c

```diff
@@ -1462,40 +1478,64 @@ __wt_find_lookaside_upd(
          */
         if (upd_type == WT_UPDATE_MODIFY) {
             WT_NOT_READ(modify, true);
+            __find_birthmark_update(cbt, &birthmark_upd);
             while (upd_type == WT_UPDATE_MODIFY) {
-                WT_ERR(__wt_update_alloc(session, las_value, &upd, &notused, upd_type));
-                if (i >= WT_MODIFY_ARRAY_SIZE) {
-                    if (i == WT_MODIFY_ARRAY_SIZE)
-                        listp = NULL;
-                    WT_ERR(__wt_realloc_def(session, &allocated_bytes, i + 1, &listp));
-                    if (i == WT_MODIFY_ARRAY_SIZE)
-                        memcpy(listp, list, sizeof(list));
-                }
-                listp[i++] = upd;
-                WT_ERR(cursor->prev(cursor));
-#ifdef HAVE_DIAGNOSTIC
+                WT_ERR(__wt_update_alloc(session, las_value, &mod_upd, &notused, upd_type));
+                WT_ERR(__wt_modify_vector_push(&modifies, mod_upd));
+                mod_upd = NULL;
+                WT_ERR(las_cursor->prev(las_cursor));
```

Comment:
Another spot where I'm wondering if `WT_NOTFOUND` is a boundary condition rather than an error.

Reply:
e03abe2